code/src/java/pcgen/rules/context/AbstractReferenceContext.java

```diff
@@ -68,7 +69,26 @@
 import pcgen.util.Logging;
 import pcgen.util.StringPClassUtil;
 
-public abstract class AbstractReferenceContext implements ObjectDatabase
+/**
```

Comment:
You realise you've re-created invokedynamic here ;-).

Reply:
Umm, no, not really (not to mention most of this code predated Java 7 ;) ).  

Can we get a benefit out of that observation?  Does it make sense to have some portion of our code come out of Java and into something that can actually produce the invokedynamic instruction?  Would we reduce code by 4X or more doing that? (or be able to get some better function - since even as complex as AbstractReferenceContext is, there are still things we want that it doesn't do?

Reply:
Yeah - we could use MethodReferences and MethodHandles that get lazily resolved.  Again a pretty major global change though.